src/net_processing.cpp

```diff
@@ -1108,6 +1116,71 @@ class PeerManagerImpl final : public PeerManager
             }
         }
 
+        /**
+         * Get the transaction that has been broadcast less times and least recently.
+         */
+        std::optional<CTransactionRef> GetTxForBroadcast() const EXCLUSIVE_LOCKS_REQUIRED(!m_mutex)
+        {
+            LOCK(m_mutex);
+            if (m_by_priority.empty()) {
+                return std::nullopt;
+            }
+            const Txid& txid = m_by_priority.begin()->second;
+            auto it = m_by_txid.find(txid);
+            if (it == m_by_txid.end()) {
+                return std::nullopt;
```

Comment:
07dee91112189be02c6694385de261570725b56b this would be unexpected if we had the txid in the priority list but not in the actual txid list, right? Maybe we should at least remove the txid from the priority list if this ever happens, or log something, or like, call the police or something?

Reply:
Added `Assume()` that the entry is found.